File: src/main/java/net/shotbow/interestingfish/listeners/InteractListener.java
package net.shotbow.interestingfish.listeners;

import net.shotbow.interestingfish.config.InterestingConfig;
import org.bukkit.Tag;
import org.bukkit.entity.ItemFrame;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerInteractEntityEvent;
import org.bukkit.inventory.ItemStack;

/**
 * User: Austin
 * Date: 5/18/13
 * Time: 1:47 AM
 * (c) lazertester
 */
public class InteractListener implements Listener {

    private final InterestingConfig config;

    public InteractListener(InterestingConfig config) {
        this.config = config;
    }

    @EventHandler
    public void onFish(PlayerInteractEntityEvent e) {
        if (config.showItemFrameInfo && e.getRightClicked() instanceof ItemFrame) {
            final ItemStack item = ((ItemFrame) e.getRightClicked()).getItem();
            if (item != null && Tag.ITEMS_FISHES.isTagged(item.getType()) && !e.getPlayer().isSneaking()) {
                e.setCancelled(true);
                final Player player = e.getPlayer();
                if (item.hasItemMeta()) {
                    if (item.getItemMeta().hasDisplayName()) {
                        player.sendMessage(item.getItemMeta().getDisplayName());
                    }
                    if (item.getItemMeta().hasLore()) {
                        for (String lore : item.getItemMeta().getLore()) {
                            player.sendMessage(lore);
                        }
                    }
                }
            }
        }
    }
}

File: src/main/kotlin/net/shotbow/interestingfish/listeners/InteractListener.kt
package net.shotbow.interestingfish.listeners

import net.shotbow.interestingfish.config.InterestingConfig
import net.shotbow.interestingfish.util.ItemUtility.isFish
import org.bukkit.entity.ItemFrame
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerInteractEntityEvent

class InteractListener(private val config: InterestingConfig) : Listener {
    @EventHandler
    fun onFish(e: PlayerInteractEntityEvent) {
        val rightClicked = e.rightClicked

        if (config.showItemFrameInfo && rightClicked is ItemFrame) {
            val item = rightClicked.item
            if (item.isFish() && !e.player.isSneaking) {
                e.isCancelled = true

                if (item.hasItemMeta()) {
                    val itemMeta = item.itemMeta!!
                    if (itemMeta.hasDisplayName()) {
                        e.player.sendMessage(itemMeta.displayName)
                    }
                    if (itemMeta.hasLore()) {
                        for (lore in itemMeta.lore!!) {
                            e.player.sendMessage(lore)
                        }
                    }
                }
            }
        }
    }
}
